### PR TITLE
fix: Remove dead transactions consumers code

### PR DESCRIPTION
**Pull request by untitaker**
This code was never used as the message type for transactions is currently also `event`. The other code is used in event manager, and there it's somewhat useful to have dedicated codepath for transactions, even though the batching isn't part of why it's useful.

I have checked that the metrics `process_transactions_batch` is not seen in prod

### CODE REVIEW

File: src/sentry/ingest/ingest_consumer.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from sentry.attachments import CachedAttachment, attachment_cache
 from sentry.ingest.types import ConsumerType
 from sentry.ingest.userreport import Conflict, save_userreport
-from sentry.event_manager import save_attachment, save_transaction_events
+from sentry.event_manager import save_attachment
```
Comment (jan-auer) [inline src/sentry/ingest/ingest_consumer.py:27]: Can you also remove `save_transaction_events`, then?

**Review by jan-auer — APPROVED** [review]

File: src/sentry/ingest/ingest_consumer.py (hunk 1 above)
Comment (untitaker) [reply src/sentry/ingest/ingest_consumer.py:27]: This is used inside of event_manager and is basically the "separate codepath" I mentioned in the PR description. I *think* we would want to keep this even though maybe no longer batching, not going through the regular event manager save still sounds nice to have.

**Review by beezz — CHANGES_REQUESTED** [review]
Let's also remove the `store.transactions-celery` option: 
- https://github.com/getsentry/sentry/blob/master/src/sentry/options/defaults.py#L197-L199
- https://github.com/getsentry/sentry/blob/master/tests/sentry/ingest/ingest_consumer/test_ingest_consumer_kafka.py#L106

**Comment by untitaker** [comment]
actual duplicate of https://github.com/getsentry/sentry/pull/17578/ but let's go with this one

**Review by beezz — APPROVED** [review]

**Pull request merged**